doc/vfd-swmr-user-guide.md

```diff
@@ -0,0 +1,496 @@
+Instead, the reader will see arbitrary values or the fill value.
+To display those values would be distracting and misleading to
+the experimenter. 
+
+On the reader, a strategy for displaying the most current, bonafide application
+data is to read the dimensions of the frames dataset, `d`, compute
+the number `n` of full frames contained in `d`, and read the
+next-to-last frame, `n - 2`.  THG uses a variant of this strategy
+in its `gaussians` demo.
+
+On the writer, a strategy for protecting against snapshots between
+the `H5Dset_extent()` and `H5Dwrite()` calls is to suspend VFD
+SWMR's clock across both of the calls.  The
+`H5Fvfd_swmr_disable_end_of_tick()` call takes a file identifier
+and stops new snapshots from being taken on the given file until
+`H5Fvfd_swmr_enable_end_of_tick()` is called on the same file.
+Needless to say, end of tick processing should only be disabled
+briefly.
+
+# Known issues
+
+## Variable-length data
+
+A VFD SWMR reader cannot reliably read back a variable-length dataset
+written by VFD SWMR.  For example, a variable-length string
+created and written as follows
+
+```
+    hid_t dset, space, type;
+    char data[] = "content";
+
+    type = H5Tcopy(H5T_C_S1);
+
+    H5Tset_size(type, H5T_VARIABLE);
+
+    space = H5Screate(H5S_SCALAR);
+
+    dset = H5Dcreate2(..., "string", type, space, H5P_DEFAULT, H5P_DEFAULT,
+        H5P_DEFAULT);
+
+    H5Dwrite(dset, type, space, space, H5P_DEFAULT, &data);
+```
+
+and read back like this,
+
+```
+    char *data;
+    herr_t ret;
+
+    ret = H5Dread(..., ..., H5S_ALL, H5S_ALL, H5P_DEFAULT, &data);
+```
+
+may produce either an error return from `H5Dread` (`ret < 0`) or
+a `NULL` pointer (`data == NULL`).
+
+As discussed above, this is caused by a fundamental incompatibility 
+between the current variable length data implementation in HDF5, which 
+stores variable length data as metadata.  It is possible we may be able 
+to mitigate the issue, but the most likely solution is the planned 
+re-implementation of variable length data that is currently in the planning
+stage.  Unfortunately, we have no ETA for this re-implementation.
+
+## Iteration
+
+An application that reads in VFD SWMR mode should take care to avoid
+HDF5 iteration APIs, especially when iterating large numbers of objects
+or using long-running application callbacks.  While the library is in an
+iteration routine, it cannot track changes made by the writer.  If the
+library spends more than `max_lag` ticks in the routine, then its view
+of the HDF5 file will become stale.  Under those circumstances, HDF5
+content could be mis-read, or the library could crash with a diagnostic
+assertion.
+
+## Object handles
+
+At the present level of development, the writer cannot invalidate
+a reader's HDF5 object handles (`hid_t`s).  If a reader holds an
+object open---that is, it has a valid handle (`hid_t`) for the
+object---while the writer deletes it, then reading content through
```

Comment:
Again, probably should disable deleting objects when the writer is using the SWMR VFD.